api/service/stagedstreamsync/downloader.go

```diff
@@ -119,8 +134,9 @@ func (d *Downloader) DownloadAsync() {
 	select {
 	case d.downloadC <- struct{}{}:
 		consensusTriggeredDownloadCounterVec.With(d.promLabels()).Inc()
-
-	case <-time.After(100 * time.Millisecond):
+	case <-d.closeC:
```

Comment:
downloadC initialized with `make(chan struct{}, 1)`, so it could be send even closed

Reply:
seems like minor